### PR TITLE
MEN-5012: Remove `PACKAGECONFIG = "dbus"` for new Mender versions.

### PR DESCRIPTION
**Pull request by kacf**
It is no longer possible to compile the Mender client without dbus
support, and glib-2.0 is now a required dependency.

Changelog: Commit

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>

### CODE REVIEW

File: meta-mender-core/recipes-mender/mender-client/mender-client.inc [hunk 1]
```diff
@@ -272,23 +267,17 @@ do_install() {
...
+    oe_runmake \
+        -C ${B}/src/${GO_IMPORT} \
+        V=1 \
+        prefix=${D} \
+        datadir=${datadir} \
+        install-dbus
+
+    # install the D-Bus interface file(s)
+    install -d ${D}${datadir}/dbus-1/interface
+    install -m 0644 ${B}/src/${GO_IMPORT}/Documentation/io.mender.*.xml ${D}${datadir}/dbus-1/interface
```
Comment (oleorhagen) [inline meta-mender-core/recipes-mender/mender-client/mender-client.inc:280]: Question: Should we consider adding this to the `Makefile`?

**Review by oleorhagen — APPROVED** [review]

File: meta-mender-core/recipes-mender/mender-client/mender-client_2.5.0.bb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-require mender-client.inc
+require mender-client-pre-3.1.inc
```
Comment (lluiscampos) [inline meta-mender-core/recipes-mender/mender-client/mender-client_2.5.0.bb:1]: It should be pre 3.2, right? Mender client 3.1 we are tentatively releasing in few weeks from now, and this feature goes into the following one, 3.2.

File: tests/acceptance/test_build.py [hunk 3]
```diff
@@ -1251,40 +1214,29 @@ def test_mender_dbus_interface_file(
...
-                bitbake_image,
-                target="-c clean mender-client",
-            )
+        # clean up the mender-client
+        build_image(
+            prepared_test_build["build_dir"],
+            prepared_test_build["bitbake_corebase"],
+            bitbake_image,
+            target="-c clean mender-client",
+        )
```
Comment (lluiscampos) [inline tests/acceptance/test_build.py:1223]: I don't think we need this clean call anymore.

**Review by lluiscampos — APPROVED** [review]

File: meta-mender-core/recipes-mender/mender-client/mender-client_2.5.0.bb (hunk 2 above)
Comment (kacf) [reply meta-mender-core/recipes-mender/mender-client/mender-client_2.5.0.bb:1]: Yes! Good catch!

File: tests/acceptance/test_build.py (hunk 3 above)
Comment (kacf) [reply tests/acceptance/test_build.py:1223]: Probably not, but I think it's worth having as a safeguard anyway. A bit like doing `rm logfile && echo "first entry" > logfile`.

File: meta-mender-core/recipes-mender/mender-client/mender-client.inc (hunk 1 above)
Comment (kacf) [reply meta-mender-core/recipes-mender/mender-client/mender-client.inc:280]: Good idea, I will add it.

**Pull request merged**